src/stats.rs

```diff
@@ -58,14 +63,61 @@ impl fmt::Debug for Distribution {
     }
 }
 
-impl fmt::Display for Distribution {
+impl fmt::Display for Aggregator {
     fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
         writeln!(fmt, "Min: {:6}", self.min)?;
         writeln!(fmt, "Max: {:6}", self.max)?;
         writeln!(fmt, "Avg: {:6.2}", self.avg)
     }
 }
 
+pub struct Distribution(BTreeMap<u64, u64>);
+
+impl Distribution {
+    pub fn new<I>(values: I) -> Self
+    where
+        I: IntoIterator<Item = u64>,
+    {
+        let mut map = BTreeMap::new();
+
+        for value in values {
+            *map.entry(value).or_insert(0) += 1;
+        }
+
+        Distribution(map)
+    }
+
+    pub fn summary(&self) -> Aggregator {
+        if self.0.is_empty() {
+            return Aggregator::empty();
+        }
+
+        let mut avg = 0.0;
+        let mut num = 0;
+
+        for (&key, &value) in &self.0 {
+            avg += key as f64 * value as f64;
```

Comment:
do we need to use `f64::from` here?

Reply:
There is no `impl From<u64> for f64` at the moment, so no.